src/jdk.jdwp.agent/share/native/libjdwp/eventHandler.c

```diff
@@ -681,23 +681,14 @@ event_callback(JNIEnv *env, EventInfo *evinfo)
         bagDestroyBag(eventBag);
     }
 
-    /* Always restore any exception that was set beforehand.  If
-     * there is a pending async exception, StopThread will be
-     * called from threadControl_onEventHandlerExit immediately
-     * below.  Depending on VM implementation and state, the async
-     * exception might immediately overwrite the currentException,
-     * or it might be delayed until later.  */
-    if (currentException != NULL) {
-        JNI_FUNC_PTR(env,Throw)(env, currentException);
-    } else {
-        JNI_FUNC_PTR(env,ExceptionClear)(env);
-    }
+    /* Clear any exception thrown while handling the event. */
+    JNI_FUNC_PTR(env,ExceptionClear)(env);
```

Comment:
For those doing the review, the rethrowing of the exception has moved into threadControl_onEventHandlerExit(), which is called below. The rethrowing needs to occur after the last findRunningThread() call since that triggers a call to JNI IsVirtualThread(), which shouldn't be called with a pending exception.